content/en/docs/generating-objects/instantiate-methods.md

```diff
@@ -0,0 +1,644 @@
+{{< tab header="Java" lang="java">}}
+
+@Test
+void test() {
+    Product product = fixtureMonkey.giveMeBuilder(Product.class)
+        .instantiate(constructor())
+        .sample();
+
+    then(product.getId()).isEqualTo(0);
+}
+
+{{< /tab >}}
+{{< tab header="Kotlin" lang="kotlin">}}
+
+@Test
+fun test() {
+    val product = fixtureMonkey.giveMeBuilder<Product>()
+        .instantiateBy {
+            constructor()
+        }
+        .sample()
+
+    then(product.productName).isEqualTo("")
+}
+
+{{< /tab >}}
+{{< /tabpane>}}
+You can specify to use the constructor method by passing the `constructor()` option.
+If there is a NoArgsConstructor, it will be used to create the object. If not, the first written constructor is used.
+
+### Specifying a Constructor
+If a class has multiple constructors, you can specify the desired constructor by providing the necessary parameter information.
+For instance, consider the Product class with two constructors requiring different parameters.
+
+To use the constructor that creates instances with empty options, specify the paramters as follows:
+
+{{< tabpane persist=false >}}
+{{< tab header="Java" lang="java">}}
+
+@Test
+void test() {
+    Product product = fixtureMonkey.giveMeBuilder(Product.class)
+        .instantiate(
+            constructor()
+                .parameter(String.class)
+                .parameter(long.class)
+                .parameter(long.class)
+        )
+        .sample();
+
+    then(product.getOptions()).isEmpty();
+}
+
+{{< /tab >}}
+{{< tab header="Kotlin" lang="kotlin">}}
+
+@RepeatedTest(TEST_COUNT)
+fun test() {
+    val product = fixtureMonkey.giveMeBuilder<Product>()
+        .instantiateBy {
+            constructor<Product> {
+                parameter<String>()
+                parameter<Long>()
+                parameter<Long>()
+        }
+    }
+    .sample()
+
+    then(product.options).isEmpty()
+}
+
+{{< /tab >}}
+{{< /tabpane>}}
+
+If you want to use the other constructor that makes a `Product` with productName as "defaultProductName", you can specify it by just changing the parameter information:
+{{< tabpane persist=false >}}
+{{< tab header="Java" lang="java">}}
+
+constructor()
+    .parameter(long.class)
+    .parameter(long.class)
+    .parameter(new TypeReference<List<String>>(){})
+
+{{< /tab >}}
+{{< tab header="Kotlin" lang="kotlin">}}
+
+constructor<Product> {
+    parameter<Long>()
+    parameter<Long>()
+    parameter<List<String>>()
+}
+
+{{< /tab >}}
+{{< /tabpane>}}
+
+Note that using private constructors is also possible.
+
+### Constructor Parameter Hints
+In cases where you want to pass a specific value to the constructor, you can add a parameter name hint.
+
+{{< tabpane persist=false >}}
+{{< tab header="Java" lang="java">}}
+
+@Test
+void test() {
+    Product product = fixtureMonkey.giveMeBuilder(Product.class)
+        .instantiate(
+            constructor()
+                .parameter(String.class, "str")
```

Comment:
You can set the hint to anything you want, but we like it to be the name of the constructor parameter in a official document. It may cause confusion.